docs/madr/decisions/089-naming-envoy-resources-and-stats-from-multiple-kuma-policies.md

```diff
@@ -0,0 +1,249 @@
+
+| Scenario                    | Merge Location           | Container Naming | Component Naming     | Stats Attribution               |
+|:----------------------------|:-------------------------|:-----------------|:---------------------|:--------------------------------|
+| Multi-MeshHTTPRoute         | RouteConfig.routes[]     | Service KRI      | Policy KRI per route | Routes: ✅ HCM stats: aggregate |
+| Multi-MeshTrafficPermission | RBAC filter rules        | Contextual       | N/A                  | Aggregate                       |
+| Multi-MeshTimeout           | route.timeout            | Service KRI      | Inherits route       | Aggregate at HCM level          |
+| Multi-MeshRetry             | route.retry_policy       | Service KRI      | Inherits route       | Aggregate at HCM level          |
+| Multi-MeshHealthCheck       | cluster.health_checks[]  | Service KRI      | N/A                  | Aggregate                       |
+| Multi-MeshCircuitBreaker    | cluster.circuit_breakers | Service KRI      | N/A                  | Aggregate                       |
+
+### Example: Three MeshHTTPRoute Policies Merged
+
+```yaml
+Listener:
+  name: kri_msvc_default_zone-1_demo-app_backend-api_httpport
+  filter_chains:
+    - filters:
+        - name: envoy.filters.network.http_connection_manager
+          typed_config:
+            stat_prefix: kri_msvc_default_zone-1_demo-app_backend-api_httpport  # Aggregate
+            route_config:
+              routes:
+                - name: kri_mhttpr_default_zone-1_demo-app_backend-routing_rule_0      # Policy 1
+                - name: kri_mhttpr_default__kuma-system_client-to-backend_rule_0       # Policy 2
+                - name: kri_mhttpr_default_zone-1_demo-app_backend-with-timeout_rule_0 # Policy 3
+```
+
+**Stats interpretation:**
+
+- `http.kri_msvc_..._backend-api_httpport.downstream_rq_2xx` → Aggregate across all three policies
+- Access log `route_name: kri_mhttpr_..._backend-routing_rule_0` → Specific policy attribution
+
+## Design Options
+
+| Option                      | Approach                                              | Verdict                                  |
+|:----------------------------|:------------------------------------------------------|:-----------------------------------------|
+| Encode all policies         | Include all policy KRIs in resource name              | ❌ Cardinality explosion, unstable names |
+| Multi-policy marker         | Add `__multipolicy` suffix                            | ❌ No actual attribution benefit         |
+| Status quo                  | Keep current behavior                                 | ❌ Maintains traceability gaps           |
+| **Aggregate + Attribution** | Service KRI for containers, policy KRI for components | ✅ Recommended                           |
+
+### Recommended: Aggregate Naming + Policy Attribution
+
+**Approach:**
+
+1. **Container resources** (listeners, clusters, HCM): Use service KRI or contextual format
```

Comment:
Container resources - what are container resources? It's not fully clear for me naming here. Could we name it better as it seems more like xds configurations ?